File: src/main/java/game/actors/Player.java
package game.actors;

import requests.spark.websockets.objects.JSONWeb;
import requests.spark.websockets.objects.messages.mapping.WSData;
import requests.spark.websockets.objects.messages.mapping.WSDataReference;
import requests.spark.websockets.objects.messages.mapping.WSDataTypeScriptClass;
import utils.managers.PlayerCounterManager;

import java.util.UUID;

public class Player extends JSONWeb {
    @WSDataReference(WSData.PLAYER_UUID)
    private UUID uuid;
    @WSDataReference(WSData.PLAYER_ID)
    private Integer id;
    @WSDataReference(WSData.PLAYER_NAME)
    private String name;

    @WSDataReference(WSData.PLAYER_SCORE)
    private Integer score;
    @WSDataReference(WSData.PLAYER_PLAYER_STATUS)
    private String playerStatus;

    @WSDataReference(WSData.PLAYER_LATEST_ANSWER)
    @WSDataTypeScriptClass(Answer.class)
    private Answer latestAnswer;

    public Player() {
        uuid = UUID.randomUUID();
        id = PlayerCounterManager.getInstance().getNextPlayerID();
        name = "Player " + id;
        score = 0;
        playerStatus = "alert-light";
        latestAnswer = null;
    }

    public UUID getUuid() {
        return uuid;
    }

    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setUuid(UUID uuid) {
        this.uuid = uuid;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getScore() {
        return score;
    }

    public void setScore(Integer score) {
        this.score = score;
    }

    public String getPlayerStatus() {
        return playerStatus;
    }

    public void setPlayerStatus(String playerStatus) {
        this.playerStatus = playerStatus;
    }

    public Answer getLatestAnswer() {
        return latestAnswer;
    }

    public void setLatestAnswer(Answer latestAnswer) {
        this.latestAnswer = latestAnswer;
    }
}

File: src/main/ngapp/src/app/player.ts
import {Answer} from "./answer";

export class Player {
  private _uuid: string;
  private _latestAnswer: Answer;
  private _id: number;
  private _score: number = 0;
  private _name: string;
  private _playerStatus: string = "alert-light";

  constructor(uuid: string) {
    this._uuid = uuid;
  }

  get latestAnswer(): Answer {
    return this._latestAnswer;
  }

  set latestAnswer(value: Answer) {
    this._latestAnswer = value;
  }

  get uuid(): string {
    return this._uuid;
  }

  set uuid(value: string) {
    this._uuid = value;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get score(): number {
    return this._score;
  }

  set score(value: number) {
    this._score = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get playerStatus(): string {
    return this._playerStatus;
  }

  set playerStatus(value: string) {
    this._playerStatus = value;
  }
}
